schemas/2015-06-01/Microsoft.KeyVault.json

```diff
@@ -0,0 +1,209 @@
+{
+    "id": "http://schema.management.azure.com/schemas/2015-06-01/Microsoft.KeyVault.json#",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Microsoft.KeyVault",
+    "description": "Microsoft KeyVault Resource Types",
+    "definitions": {
+        "keyPermission": {
+            "anyOf": [
+                {
+                    "enum": [
+                        "all"
+                    ]
+                },
+                {
+                    "enum": [
+                        "backup",
+                        "create",
+                        "decrypt",
+                        "delete",
+                        "encrypt",
+                        "get",
+                        "import",
+                        "list",
+                        "restore",
+                        "sign",
+                        "unwrapKey",
+                        "update",
+                        "verify",
+                        "wrapKey"
+                    ]
+                }
+            ]
+        },
+        "secretPermission": {
+            "anyOf": [
+                {
+                    "enum": [
+                        "all"
+                    ]
+                },
+                {
+                    "enum": [
+                        "delete",
+                        "get",
+                        "list",
+                        "set"
+                    ]
+                }
+            ]
+        },
+        "permissions": {
+            "type": "object",
+            "properties": {
+                "keys": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/keyPermission"
+                    },
+                    "uniqueItems": true,
+                    "minItems": 1,
+                    "description": "Microsoft.KeyVault/vaults: The permissions granted on keys in this vault to this Active Directory object."
+                },
+                "secrets": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/secretPermission"
+                    },
+                    "uniqueItems": true,
+                    "minItems": 1,
+                    "description": "Microsoft.KeyVault/vaults: The permissions granted on secrets in this vault to this Active Directory object."
+                }
+            },
+            "minProperties": 1
```

Comment:
Thanks - I wasn't familiar with additionalProperties.  It looks like I should be using this a lot more (for example, it looks like using this would have caught the error you found below with enabledForDeployment).